File: RDALoggerLibrary/src/main/java/com/ardakaplan/rdalogger/LogType.java
package com.ardakaplan.rdalogger;

import android.util.Log;

/**
 * Created by Arda Kaplan at 4.08.2021 - 14:24
 * <p>
 * ardakaplan101@gmail.com
 */
public enum LogType {

    VERBOSE(Log.VERBOSE),
    DEBUG(Log.DEBUG),
    INFO(Log.INFO),
    WARN(Log.WARN),
    ERROR(Log.ERROR),
    //    ASSERT(Log.ASSERT),
    LIFE_CYCLE(-1);

    private final int priority;

    LogType(int priority) {
        this.priority = priority;
    }

    public int getPriority() {
        return priority;
    }
}

File: RDALogger/src/main/java/com/ardakaplan/library/rdalogger/LogType.kt
package com.ardakaplan.library.rdalogger

import android.util.Log
import androidx.annotation.IntDef

/**
 * Created by Arda Kaplan on 21/02/2025 at 14:42.
 *
 * ardakaplan101@gmail.com
 */

enum class LogType(@Priority val priority : Int) {

    VERBOSE(Log.VERBOSE),
    DEBUG(Log.DEBUG),
    INFO(Log.INFO),
    WARN(Log.WARN),
    ERROR(Log.ERROR),
    LIFE_CYCLE(-1);

    companion object {
        @Retention(AnnotationRetention.SOURCE)
        @IntDef(Log.ASSERT, Log.ERROR, Log.WARN, Log.INFO, Log.DEBUG, Log.VERBOSE)
        annotation class Priority
    }
}